lib/modules/manager/hermit/artifacts.ts

```diff
@@ -0,0 +1,254 @@
+    },
+    { concurrency: 5 }
+  );
+
+  const deleted = hermitChanges.deleted.map(getDeleteResult);
+
+  const modified = await pMap(
+    hermitChanges.modified,
+    async (path: string): Promise<UpdateArtifactsResult[]> => {
+      const contents = await getContent(path);
+      return [
+        getDeleteResult(path), // delete existing link
+        getAddResult(path, contents), // add a new link
+      ];
+    },
+    { concurrency: 5 }
+  );
+
+  const renamed = await pMap(
+    hermitChanges.renamed,
+    async (renamed): Promise<UpdateArtifactsResult[]> => {
+      const from = renamed.from;
+      const to = renamed.to;
+      const toContents = await getContent(to);
+
+      return [getDeleteResult(from), getAddResult(to, toContents)];
+    },
+    { concurrency: 5 }
+  );
+
+  return [
+    // rename will need to go first, because
+    // it needs to create the new link for the new version
+    // for the modified links to use
+    ...flattern(renamed),
+    ...flattern(modified),
+    ...added,
+    ...deleted,
+  ];
+}
+
+/**
+ * getHermitPackage returns the hermit package for running the hermit install
+ */
+function getHermitPackage(name: string, version: string): string {
+  return `${name}-${version}`;
+}
+
+/**
+ * getHermitPackageReferenceFile returns the hermit package reference
+ * file with the given package name and version
+ */
+function getHermitPackageReferenceFile(
+  pkgName: string,
+  version: string
+): string {
+  return `bin/.${getHermitPackage(pkgName, version)}.pkg`;
+}
+
+/**
+ * updateHermitPackage runs hermit install for the given package
+ */
+async function updateHermitPackage(
+  pkg: HermitPackageDependency
+): Promise<void> {
+  logger.trace({ pkg }, `hermit.updateHermitPackage()`);
+  if (!pkg.depName || !pkg.currentVersion || !pkg.newValue) {
+    logger.error(
+      {
+        depName: pkg.depName,
+        currentVersion: pkg.currentVersion,
+        newValue: pkg.newValue,
+      },
+      'missing package update information'
+    );
+
+    throw {
+      stderr: 'invalid package to update',
+      from: pkg.currentVersion,
+      to: pkg.newValue,
+    } as UpdateHermitError<string>;
+  }
+  const depName = pkg.depName;
+  const currentVersion = pkg.currentVersion;
+  const newValue = pkg.newValue;
+  const from = getHermitPackageReferenceFile(depName, currentVersion);
+  const to = getHermitPackageReferenceFile(depName, newValue);
+  const fromPackage = getHermitPackage(depName, currentVersion);
+  const toPackage = getHermitPackage(depName, newValue);
+  const execOptions: ExecOptions = {
+    docker: {
+      image: 'sidecar',
+    },
+    cwdFile: pkg.packageFileName,
+  };
+  const execCommands = `./hermit install ${toPackage}`;
+  logger.debug(
+    {
+      fromPackage,
+      toPackage,
+      pkgName: pkg.depName,
+      packageFile: pkg.packageFileName,
+    },
+    `performing updates`
+  );
+
+  try {
+    const result = await exec(execCommands, execOptions);
```

Comment:
your should combine those exec command to a single exec, so we only start one sidecar container. otherwise we've a huge overhead for creating docker child containers.